src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
+            GAMMA_POOL_ADDRESS,
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-      approveWithdraw(depPool, depPool.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
     }
+  }, [provider])
+
+  async function approveTransaction() {
+    let loading = notifyLoading('Waiting for block confirmation')
+    approveWithdraw()
+      .then(() => {
+        setEnableRemove(true)
+        notifyDismiss(loading)
+      })
+      .catch((err) => {
+        notifyDismiss(loading)
+        setEnableRemove(false)
+        notifyError(err.toString())
+      })
   }
 
   async function withdrawLiquidity(balance: number) {
+    if (balance === 0) {
+      throw new Error('Please select a valid amount to withdraw')
+    }
     let amt = '0'
     if (balance === 100) {
       amt = totalLiquidityAmt.toString()
     } else {
       amt = ethers.utils.parseEther(((liquidityAmt * balance) / 100).toString()).toString()
     }
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
-    if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo && accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        if (accountInfo && accountInfo?.address) {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        }
-      }
-    } else {
-      notifyError('Please connect wallet')
-    }
-
     if (!accountInfo || !accountInfo.address) {
       notifyError('Wallet not connected.')
       return
     }
 
-    if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-      if (posManager === null) {
-        return
-      }
+    if (parseInt(amt) === 0) {
+      throw new Error('User does not have enough reserves to withdraw')
+    }
 
-      try {
-        const WithdrawReservesParams = {
-          cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
-          protocol: 1,
-          amount: amt,
-          amountsMin: [100, 200, 300],
-          to: accountInfo.address,
-          deadline: ethers.constants.MaxUint256,
-        }
-        let tx = await posManager.withdrawReserves(WithdrawReservesParams)
-        return await tx.wait()
-      } catch (e) {
-        return e
+    try {
+      if (!posManager) {
+        await initializePositionManagerContract()
```

Comment:
Is it possible that after this call, posManager is still null?

Reply:
Even if after this posManager is null. I have a condition that checks if we have posManager then it proceeds with the transaction